pkg/cmd/update/update.go

```diff
@@ -0,0 +1,453 @@
+			}
+
+			opts.streams = userio.NewIOStreamsWithInteractive(
+				os.Stdin,
+				os.Stdout,
+				!nonInteractive,
+			)
+
+			return update(opts)
+		},
+	}
+
+	updateCmd.Flags().BoolVar(
+		&opts.skipConfirmation,
+		"skip-confirmation",
+		false,
+		"Auto approve confirmation prompt for update.",
+	)
+
+	return updateCmd
+}
+
+func updateAvailable(githubClient *github.Client) (bool, string, error) {
+	release, err := getLatestCorectlRelease(githubClient)
+	if err != nil {
+		return false, "", err
+	}
+	asset, err := getLatestCorectlAsset(release)
+	if err != nil {
+		return false, "", err
+	}
+
+	if version.Version == asset.Version {
+		return false, "", nil
+	} else {
+		return true, asset.Version, nil
+	}
+}
+
+func update(opts UpdateOpts) error {
+	if opts.targetVersion != "" {
+		log.Debug().Msgf("target version set to %s", opts.targetVersion)
+	}
+
+	wizard := opts.streams.Wizard("Checking for updates", "Retrieved version metadata")
+	githubClient := github.NewClient(nil)
+	if opts.githubToken != "" {
+		githubClient = githubClient.WithAuthToken(opts.githubToken)
+	}
+
+	var release *github.RepositoryRelease
+	var err error
+	if opts.targetVersion == "" {
+		release, err = getLatestCorectlRelease(githubClient)
+	} else {
+		release, err = getCorectlReleaseByTag(githubClient, opts.targetVersion)
+	}
+	if err != nil {
+		wizard.Abort(err.Error())
+		return err
+	}
+
+	asset, err := getLatestCorectlAsset(release)
+	if err != nil {
+		wizard.Abort(err.Error())
+		return err
+	}
+	log.Debug().Str("current_version", version.Version).Str("remote_version", asset.Version).Msg("comparing versions")
+	if version.Version == asset.Version {
+		wizard.SetCurrentTaskCompletedTitle(fmt.Sprintf("Already running %s release (%v)", opts.targetVersion, version.Version))
+		wizard.Done()
+		return nil
+	} else {
+		wizard.SetCurrentTaskCompletedTitle(fmt.Sprintf("Update available: %v", asset.Version))
+		wizard.Done()
+	}
+
+	out, err := glamour.Render(asset.Changelog, "dark")
+	if err == nil {
+		_, _ = opts.streams.GetOutput().Write([]byte(out))
+	} else {
+		log.Warn().Err(err).Msg("could not render changelog markdown, falling back to plaintext")
+		_, _ = opts.streams.GetOutput().Write([]byte(asset.Changelog))
+	}
+
+	log.Debug().Bool("skipConfirmation", opts.skipConfirmation).Msg("checking params")
+
+	wizard = opts.streams.Wizard("Confirming update", "Confirmation received")
+	defer wizard.Done()
+	if opts.skipConfirmation {
+		wizard.Info("--skip-confirmation is set, continuing with update")
+	} else {
+		if opts.streams.IsInteractive() {
+			confirmation, err := confirmation.GetInput(opts.streams, fmt.Sprintf("Update to %s now?", asset.Version))
+			if err != nil {
+				return fmt.Errorf("could not get confirmation from user: %+v", err)
+			}
+
+			if confirmation {
+				wizard.Info("Update accepted")
+			} else {
+				err = fmt.Errorf("update cancelled by user")
+				wizard.Abort(err.Error())
```

Comment:
it panics if I do not confirm the update

Reply:
panic is due to the other check we added: `{"time":"2024-10-22T13:55:00.948+01:00","level":"panic","message":"handler is already completed"}` will fix!